jaeger-core/src/main/java/com/uber/jaeger/samplers/RemoteControlledSampler.java

```diff
@@ -148,6 +148,17 @@ public boolean equals(Object sampler) {
   }
 
   @Override
+  public int hashCode() {
```

Comment:
if the checker cannot be disabled, I am inclined to agree, we can simply rename equals and define it on the Sampler interface to be required.